neuro_san/service/agent_server.py

```diff
@@ -106,48 +110,82 @@ def setup_tool_factory_provider(self):
         for agent_name, tool_registry in self.tool_registries.items():
             tool_factory_provider.add_agent_tool_registry(agent_name, tool_registry)
 
+    def agent_added(self, agent_name: str):
+        """
```

Comment:
AgentServer itself implements AgentStateListener interface, so whenever ServiceToolFactory state (internal agents table) is changing, we react to this by talking to DynamicAgentRouter to add or remove gRPC endpoint for corresponding agent.